metron-analytics/metron-statistics/src/test/java/org/apache/metron/statistics/StellarStatisticsFunctionsTest.java

```diff
@@ -356,6 +357,44 @@ public void testSkewness() throws Exception {
     assertEquals(stats.getSkewness(), (Double) actual, 0.1);
   }
 
+  @Test
+  public void testStatsBin() throws Exception {
+    statsInit(windowSize);
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.QUARTILE.split);
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.QUARTILE.split, "'QUARTILE'");
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.QUINTILE.split, "'QUINTILE'");
+    statsBinRunner(StellarStatisticsFunctions.Bin.BinSplits.DECILE.split, "'DECILE'");
+    statsBinRunner(ImmutableList.of(25.0, 50.0, 75.0), "[25.0, 50.0, 75.0]");
+  }
+
+  public void statsBinRunner(List<Double> splits) throws Exception {
+    statsBinRunner(splits, null);
+  }
+
+  public void statsBinRunner(List<Double> splits, String splitsName) throws Exception {
+    int bin = 0;
+    for(Double d : stats.getSortedValues()) {
+      StatisticsProvider provider = (StatisticsProvider)variables.get("stats");
+      if(bin < splits.size()) {
+        double percentileOfBin = provider.getPercentile(splits.get(bin));
+        if (d > percentileOfBin) {
```

Comment:
We are walking the set of samples in sorted order and keeping track of the bin that the value is in as we walk.  When we move from one value to the other, we need to check the percentile associated with the current bin.  If it is larger than the current percentile, then we find the proper bin.

Take for example the following situation.  We will consider bounds `10` and `20`.  The  percentile mapping is as follows:
* 10th percentile is at point `17`
* 20th percentile is at point `50`

Let's take the values (in sorted order): `10`, `20`, `75`

For `10` we start with a bin of `0`.  We compare it to our current bin index, namely the 10th percentile and it's still less than the `17`, so it fits in bin `0`.

For `20` we are still at bin `0`, but now we compare it to the 10th percentile (`17`) and find that it is larger, so we need to find the next bin that is larger than `20`, which is bin `1` because the 20th percentile is 50 (so it's between the 10th percentile and 20th percentile)

For `75`, we compare to the 20th percentile (because we're at bin `1`) and it's larger, so we need to find the next percentile.  Unfortunately there's no other bound to go to (we're at the end), so it sticks on `2` or the length of bounds.

Because we can walk the sorted set, we can just move across the bins in sequential order as we pass the boundary marks.  This gives us the expected values for this test.

Does that clear things up?